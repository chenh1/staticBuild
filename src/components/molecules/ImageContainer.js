import React from 'react';
import Image from '../atoms/Image';

const ImageContainer = ({customClass, imageSrc, alt, caption}) => {
    return (
        <div className={customClass}>
            <Image imageSrc={imageSrc} alt={alt}/>
            {caption ? <span className="image-caption">{caption}</span> : ""}
        </div>
    )
};

export default ImageContainer;