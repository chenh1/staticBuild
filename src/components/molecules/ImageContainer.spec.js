import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import ImageContainer from './ImageContainer';

test('renders image container', () => {
    const tree = renderer.create(<ImageContainer
        customClass="mock-class"
        imageSrc="/mockSrc"
        alt="Mock alt" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders image container with caption span tag if caption prop is available', () => {
    const tree = renderer.create(<ImageContainer
        customClass="mock-class"
        imageSrc="/mockSrc"
        alt="Mock alt"
        caption="Mock caption" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});