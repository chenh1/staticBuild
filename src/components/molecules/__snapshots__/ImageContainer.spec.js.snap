// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders image container 1`] = `
<div
  className="mock-class"
>
  <img
    alt="Mock alt"
    src="/mockSrc"
  />
  
</div>
`;

exports[`renders image container with caption span tag if caption prop is available 1`] = `
<div
  className="mock-class"
>
  <img
    alt="Mock alt"
    src="/mockSrc"
  />
  <span
    className="image-caption"
  >
    Mock caption
  </span>
</div>
`;
